### DISCUSSION TITLE
(feat) configure key-auth plugin to validate only on header

### DISCUSSION THREAD

**Discussion by boubou191911** [Kong/kong#7270] [Ideas and feature requests]
Hello,

The key-auth plugin currenlty looks in both the HTTP headers and in the query parameters for finding the key. It is sometimes not recommended to pass sensitive information in query parameters because they are logged in browser and web server logs.
I think it can be interesting to be able to configure the key-auth plugin to only validate key against the HTTP header.

**Comment by Tieske** [comment]
it can also look in the body, when `config.key_in_body` is set.

Could be interesting to change that to `config.key_in` with a list of `header`, `query`, `body` entries. It would create more fine grained control over it.
Eg. current defaults would be `config.key_in=header,query`.

@boubou191911 PR's are welcome!

**Comment by b-rajesh** [comment]
@boubou191911 @Tieske flexibility is good.

However I just wanted to highlight apikey should not be treated as sensitive information. API Key is identifier(of an app) its not a secret. Even in OAuth2/OIDC flows , clientid(which is equivalent to APIKey , interchangeably called in few IDP providers) can be passed from the client apps to "identify" the app via the query param. Mostly APIKeys are used by SPA client apps to consume the resources(API) , where you can't store the apikey securely at all. 

Also I have noticed some organization security policy treat apikey as sensitive information.
